src/Nest/Search/Search/Highlighting/HighlightField.cs

```diff
@@ -47,9 +47,12 @@ public interface IHighlightField
 		[JsonProperty("boundary_chars")]
 		string BoundaryChars { get; set; }
 
-		[JsonProperty("type")]
 		HighlighterType? Type { get; set; }
 
+		[JsonProperty("type")]
+		[Obsolete("This is a temporary binary backwards compatible hack to make sure you can specify any custom highlightertype in 2.x, scheduled for removal in 5.0")]
+		string TypeHack { get; set; }
```

Comment:
Sounds too official in my mind, by naming it a hack explicitly we clearly signal our patch is a hack that will get removed as soon as we can.